lib/datadog/core/configuration/settings.rb

```diff
@@ -399,14 +434,15 @@ def initialize(*_)
           o.setter { |v| v.to_s if v }
 
           # NOTE: service also gets set as a side effect of tags. See the WORKAROUND note in #initialize for details.
-          o.default { ENV.fetch(Core::Environment::Ext::ENV_SERVICE, Core::Environment::Ext::FALLBACK_SERVICE_NAME) }
+          o.env Core::Environment::Ext::ENV_SERVICE
+          o.default Core::Environment::Ext::FALLBACK_SERVICE_NAME
 
           # There's a few cases where we don't want to use the fallback service name, so this helper allows us to get a
           # nil instead so that one can do
           # nice_service_name = Datadog.configuration.service_without_fallback || nice_service_name_default
           o.helper(:service_without_fallback) do
             service_name = service
-            service_name unless service_name.equal?(Core::Environment::Ext::FALLBACK_SERVICE_NAME)
+            service_name unless service_name === Core::Environment::Ext::FALLBACK_SERVICE_NAME
```

Comment:
Is that `===` (a.k.a match operator) intended?

Reply:
I believe it was an issue with some tests. I had to change it to make them pass. 

I reverted to using `equal?` https://github.com/DataDog/dd-trace-rb/pull/2970/commits/87d739d1526c9ef3c143946a641f9e909eeedbb7